test/new-e2e/tests/containers/k8s_test.go

```diff
@@ -1026,9 +1026,9 @@ func (suite *k8sSuite) testHPA(namespace, deployment string) {
 		suite.EventuallyWithTf(func(c *assert.CollectT) {
 			metrics, err := suite.Fakeintake.FilterMetrics(
 				"kubernetes_state.deployment.replicas_available",
-				fakeintake.WithTags[*aggregator.MetricSeries]([]string{
-					"kube_namespace:" + namespace,
-					"kube_deployment:" + deployment,
+				fakeintake.WithMatchingTags[*aggregator.MetricSeries]([]*regexp.Regexp{
+					regexp.MustCompile("^kube_namespace:" + regexp.QuoteMeta(namespace) + "$"),
+					regexp.MustCompile("^kube_deployment:" + regexp.QuoteMeta(deployment) + "$"),
```

Comment:
`fakeintake` API still supports both API `WithTags` and `WithMatchingTags`.
Why would we need to use `WithMatchingTags` here to match a plain string ?
It’s both less efficient and less readable.

Reply:
Sounds fair. I wanted to add some normalization between the different tests but indeed using plain string is better. I reverted that in the last commit